src/app/test/test-folder-tree/test-folder-tree.component.ts

```diff
@@ -1,86 +1,97 @@
-  @ViewChild('treeReference') treeReference!: jqxTreeComponent;
-  baseFolder: any = { label: 'Reports', value: '', filter: '', items: [], expanded: true, selected: true };
-  latestAddedFolder: any = this.baseFolder;
-  @Output() changeFolderEvent = new EventEmitter<any>();
+export class TestFolderTreeComponent {
+  @ViewChild('tree') tree!: NgSimpleFileTree;
+  @Output() changeFolderEvent: Subject<string> = new Subject<string>();
+  rootFolder: CreateTreeItem = {
+    name: 'Reports',
+    icon: 'assets/tree-icons/folder.svg',
+  };
+  treeOptions: FileTreeOptions = {
+    folderBehaviourOnClick: 'select',
+    expandAllFolders: true,
+    highlightOpenFolders: false,
+    autoSelectCondition: (item: CreateTreeItem) => this.autoSelectItem(item),
+  };
+  originalItems?: CreateTreeItem[];
 
-  constructor() {}
-
-  ngAfterViewInit(): void {
-    this.treeReference.createComponent({ source: [this.baseFolder], height: '90%', width: '100%', allowDrag: false });
-  }
-
-  updateFolderTree(reports: any, addedFolderPath: any) {
-    this.resetFolders();
-    reports.forEach((report: any) => {
-      this.addFolder(report.path);
-    });
-    this.treeReference.clear();
-    this.treeReference.addTo(this.baseFolder, null);
-    this.selectNewFolder(addedFolderPath);
+  autoSelectItem(item: CreateTreeItem): boolean {
+    return item.name === this.rootFolder.name;
   }
 
-  resetFolders() {
-    this.baseFolder = {
-      label: 'Reports',
-      value: '',
-      items: [],
-      expanded: true,
-      selected: true,
-    };
+  selectItem(value: string) {
+    value = this.removeSlashesFromPathEnds(value);
+    const path = value.length > 0 ? `Reports/${value}` : 'Reports';
+    this.tree.selectItem(path);
   }
 
-  addFolder(path: string): void {
-    if (path) {
-      let folderNames = path.startsWith('/') ? path.slice(1).split('/') : path.split('/');
-      folderNames.pop();
-      this.recursivelyAddFolders(folderNames, this.baseFolder.items, '/');
+  removeSlashesFromPathEnds(path: string) {
+    if (path.startsWith('/')) {
+      path = path.slice(1);
     }
+    if (path.endsWith('/')) {
+      path = path.slice(0, -1);
+    }
+    return path;
   }
 
-  recursivelyAddFolders(folderNames: string[], currentFolders: any[], previousFilter: string): void {
-    if (folderNames.length > 0) {
-      let name = folderNames.shift()!;
-      let currentFolder = currentFolders.find((folder) => folder.label === name);
-
-      if (!currentFolder) {
-        currentFolder = this.createNewFolder(name, previousFilter);
-        currentFolders.push(currentFolder);
-      }
-
-      this.latestAddedFolder = currentFolder;
-      this.recursivelyAddFolders(folderNames, currentFolder.items, currentFolder.value);
+  setData(data: CreateTreeItem[]) {
+    this.resetTree();
+    this.originalItems = data;
+    const tempItems = this.convertToFolders(data);
+    let rootFolder = Object.assign({}, this.rootFolder);
```

Comment:
Why a let? Its not reassigned right? 
Instead of `Object.assign({}, ...)` you could use the spread operator for shallow copying (unless this needs deep copying of course)

Reply:
You're right it should be const.

Reply:
And yes this needs to be a deep copy, thats why i've used Objects.assign